AmplifyPlugins/Geo/AWSLocationGeoPlugin/AWSLocationGeoPlugin+ClientBehavior.swift

```diff
@@ -25,51 +21,78 @@ extension AWSLocationGeoPlugin {
     ///   - countries: Limits the search to the given a list of countries/regions. (optional)
     ///   - maxResults: The maximum number of results returned per request. (optional)
     ///   - placeIndexName: The name of the Place Index to query. (optional)
-    ///   - completionHandler: The completion handler receives a Response object.  The
-    ///   success case provides a Place array.
+    /// - Returns :
+    ///     It returns a Geo.Place array.
     public func search(for text: String,
-                       options: Geo.SearchForTextOptions? = nil,
-                       completionHandler: @escaping Geo.ResultsHandler<[Geo.Place]>) {
-
-        let request = AWSLocationSearchPlaceIndexForTextRequest()!
-
+                       options: Geo.SearchForTextOptions? = nil) async throws -> [Geo.Place] {
+        
+        var request = SearchPlaceIndexForTextInput()
+        
         request.indexName = (options?.pluginOptions as? AWSLocationGeoPluginSearchOptions)?
             .searchIndex ?? pluginConfig.defaultSearchIndex
-
+        
         guard request.indexName != nil else {
-            completionHandler(.failure(.invalidConfiguration(
+            throw Geo.Error.invalidConfiguration(
                 GeoPluginErrorConstants.missingDefaultSearchIndex.errorDescription,
-                GeoPluginErrorConstants.missingDefaultSearchIndex.recoverySuggestion)))
-            return
+                GeoPluginErrorConstants.missingDefaultSearchIndex.recoverySuggestion)
         }
-
+        
         request.text = text
-
+        
         if let area = options?.area {
             switch area {
             case .near(let coordinates):
-                request.biasPosition = [coordinates.longitude as NSNumber,
-                                        coordinates.latitude as NSNumber]
+                request.biasPosition = [coordinates.longitude,
+                                        coordinates.latitude]
             case .within(let boundingBox):
-                request.filterBBox = [boundingBox.southwest.longitude as NSNumber,
```

Comment:
It makes me unreasonably happy to see that we don't have to bridge to ObjC types anymore ❤️